lib/macho/macho_file.rb

```diff
@@ -381,15 +385,16 @@ def get_load_commands
       load_commands = []
 
       header.ncmds.times do
-        cmd = @raw_data.slice(offset, 4).unpack("V").first
+        fmt = (endianness == :little) ? "L<" : "L>"
+        cmd = @raw_data.slice(offset, 4).unpack(fmt).first
         cmd_sym = LOAD_COMMANDS[cmd]
 
         raise LoadCommandError.new(cmd) if cmd_sym.nil?
 
         # why do I do this? i don't like declaring constants below
         # classes, and i need them to resolve...
         klass = MachO.const_get "#{LC_STRUCTURES[cmd_sym]}"
-        command = klass.new_from_bin(@raw_data, offset, @raw_data.slice(offset, klass.bytesize))
+        command = klass.new_from_bin(@raw_data, @endianness, offset, @raw_data.slice(offset, klass.bytesize))
```

Comment:
I think for read access we want to rely on the `endianess` reader method instead of the `@endianess` instance variable.